src/main/scala/at/logic/skeptik/expression/formula/formulas.scala

```diff
@@ -57,9 +64,41 @@ object Atom extends Formula {
     require(atom.t == o)
     atom
   }
+  def apply(name: String, args: List[E]) = {
+    def createType(list : List[E]) : T = list match {
+      case Nil   => o
+      case e::es => e.t -> createType(es)
+    }
+    val p    = Var(name,createType(args))
+    val atom = AppRec(p,args)
+    require(atom.t == o)
+    atom
+  }
   def unapply(e:E) = e match {
     case AppRec(f,args) if (e.t == o && !isLogicalConnective(f)) => Some((f,args))
     case _ => None
   }
 }
 
+object ConditionalFormula extends Formula{
```

Comment:
Just a minor style comment:  could you add spaces before the opening brackets? I mean it is better to write

`... extends Formula { `

than 

`... extends Formula{ `

It seems there are a few places where this happens...